eva-uow/src/main/kotlin/com/razz/eva/uow/composable/ChangesDsl.kt

```diff
@@ -89,13 +90,13 @@ class ChangesDsl internal constructor(initial: ChangesAccumulator) {
     suspend fun <PRINCIPAL, PARAMS, RESULT, UOW> execute(
         uow: UOW,
         principal: PRINCIPAL,
-        params: () -> PARAMS,
+        params: InstantiationContext.() -> PARAMS,
     ): RESULT
         where PRINCIPAL : Principal<*>,
               PARAMS : UowParams<PARAMS>,
               RESULT : Any,
               UOW : BaseUnitOfWork<PRINCIPAL, PARAMS, RESULT, *> {
-        val subChanges = uow.tryPerform(principal, params())
+        val subChanges = uow.tryPerform(principal, params(InstantiationContext(0)))
```

Comment:
it is safe to assume first (0th) attempt for nested uow as parent uow (and parent uow params) would be reinstantiated upon retry